frontend/src/components/VaultDetails.vue

```diff
@@ -202,18 +213,19 @@ import FetchError from './FetchError.vue';
 import GrantPermissionDialog from './GrantPermissionDialog.vue';
 import ReactivateVaultDialog from './ReactivateVaultDialog.vue';
 import RecoverVaultDialog from './RecoverVaultDialog.vue';
-import RecoveryKeyDialog from './RecoveryKeyDialog.vue';
+import DisplayRecoveryKeyDialog from './DisplayRecoveryKeyDialog.vue';
 import SearchInputGroup from './SearchInputGroup.vue';
 
 const { t, d } = useI18n({ useScope: 'global' });
 
 const props = defineProps<{
   vaultId: string,
-  role: VaultRole
+  vaultRole: 'OWNER' | 'MEMBER' | 'NONE',
```

Comment:
Is there a reason, why you didn't keep the `VaultRole` type? It's defined in `backend.ts` and can also be extended by `'NONE'`.

Reply:
Yes. 

In the backend the vaultRole `NONE` does not exist. The vaultRole field is only used in the `/vault/[vaultid]/members` endpoint and to keep the direct mapping in code, i did not touch `backend.ts`. Furthermore, in the context of the VaultList component, the role is determined indirect by checking the owned and accessible vaults and not using the VaultRole enum.

This keeps the model clean, but allows to differentiate in the GUI different cases.

Reply:
Haven't tested it, but shouldn't be `VaultRole | 'NONE'` possible in that case?

Reply:
It seems to work. Should VaultList.vue then also be adapted to use the vaultRole? Currently, there are only Strings used.
https://github.com/cryptomator/hub/blob/79d980d37177daab6df0ff4cc75b24a85da2d41b/frontend/src/components/VaultList.vue#L140-L148

Reply:
Yes, I've thought about this as well, but didn't comment on it. How would you approach this? Something like this?

```ts
const roleOfSelectedVault = computed<VaultRole | 'NONE'>(() => {
```

Reply:
Done in https://github.com/cryptomator/hub/pull/271/commits/e4204c450f20c6797560e8dc53b0aa82340bcdc5